token/program-2022/src/error.rs

```diff
@@ -156,9 +155,36 @@ pub enum TokenError {
             the associated `maximum_pending_balance_credit_counter`"
     )]
     MaximumPendingBalanceCreditCounterExceeded,
+
+    // 40
     /// The deposit amount for the confidential extension exceeds the maximum limit
     #[error("Deposit amount exceeds maximum limit")]
     MaximumDepositAmountExceeded,
+    /// CPI Guard cannot be enabled or disabled in CPI
+    #[error("CPI Guard cannot be enabled or disabled in CPI")]
+    CpiGuardSettingsLocked,
+    /// CPI Guard is enabled, and a program attempted to transfer user funds without using a delegate
+    #[error("CPI Guard is enabled, and a program attempted to transfer user funds without using a delegate")]
```

Comment:
nit: in all of these error messages, to be totally clear, can you add "from within a CPI" at the end? ie
```suggestion
    #[error("CPI Guard is enabled, and a program attempted to transfer user funds without using a delegate from within a CPI")]
```
Or some other language to that effect